// stock_note_api_documentation/src/index.js
/*
 * StockNote API Documentation
 *  StockNote API is a set of Rest APIs using which users can build customized applications based on their trading requirements. It facilitates the users of the APIs to login, search symbols, place orders and execute them, view their order status, positions and holdings etc. This documentation provides you with all the necessary details to understand the SAMCO StockNote API collection. APIs are compatible with both Javascript and Python.    For any issues or support, please raise ticket to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.      For Reference you can download postman collection <a href=\"http://developers.stocknote.com/doc/StockNoteApi.postman_collection.json\">Click here</a>      For downloading a list of all tradeable scrips across exchanges please <a href=\"https://developers.stocknote.com/doc/ScripMaster.csv\">Click here</a> This is a CSV file which you can import into your database.     If you are using Python or Java as client platform to code your strategies and integrate with our APIs, you can use our StockNoteBridge  which provides pre-written wrappers over StockNote APIs that ensure seamless and reliable connectivity to the APIs and help you fast-track your development process.  <table style=\"width:100%\"><tr><th>Languages</th><th>Documentation</th></tr><tr><td>Python</td><td><a href=\"https://github.com/samco-sdk/Python-SDK/blob/master/README.md\">Python SDK Documentation</a></td></tr><tr><td>Java</td><td><a href=\"https://github.com/samco-sdk/Java-SDK/blob/master/README.md\">Java SDK Documentation</a></td></tr></table> Add 'session token' of login api response as 'x-session-token' in header param of all Apis.      NOTE: To ensure stability and there by provide seamless services to our customers, Samco may set limits on your use of the StockNote APIs (for example, limit on the number of requests sent to a specific API) . If you have additional questions regarding the rate limits on APIs, please reach out to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.   
 *
 * OpenAPI spec version: 1.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.16
 *
 * Do not edit the class manually.
 *
 */

(function(factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/AskValues', 'model/BidAsk', 'model/BidValues', 'model/CancelOrderResponse', 'model/CandleData', 'model/EquitySearchResponse', 'model/HistoricalCandleResponse', 'model/Holding', 'model/HoldingResponse', 'model/HoldingSummary', 'model/IndexCandleData', 'model/IndexCandleDataResponse', 'model/IndexIntraDayCandleData', 'model/IndexIntraDayCandleDataResponse', 'model/IndexQuoteResponse', 'model/IntraDayCandleData', 'model/IntraDayCandleResponses', 'model/Limit', 'model/LimitResponse', 'model/LoginRequest', 'model/LoginResponse', 'model/MarketDepthResponse', 'model/ModifyOrderRequest', 'model/OptionChain', 'model/OptionChainResponse', 'model/OrderBookEntry', 'model/OrderBookResponse', 'model/OrderDetails', 'model/OrderDetailsBO', 'model/OrderDetailsCO', 'model/OrderRequest', 'model/OrderRequestBO', 'model/OrderRequestCO', 'model/OrderResponse', 'model/OrderResponseBO', 'model/OrderResponseCO', 'model/OrderStatusResponse', 'model/Position', 'model/PositionConversionRequest', 'model/PositionConversionResponse', 'model/PositionResponse', 'model/PositionSquareOffListRequest', 'model/PositionSquareOffListResponse', 'model/PositionSquareOffRequest', 'model/PositionSummary', 'model/ScripDetails', 'model/SimpleResponse', 'model/StreamingMarkDataRequest', 'model/StreamingMarkDataResponse', 'model/StreamingQuoteDataRequest', 'model/StreamingQuoteDataResponse', 'model/TradeBookEntry', 'model/TradeBookResponse', 'model/TriggerOrder', 'model/TriggerOrdersResponse', 'api/HistoricalCandleDataApi', 'api/HoldingsApi', 'api/IntraDayCandleDataApi', 'api/OptionApi', 'api/OrdersApi', 'api/PositionsApi', 'api/QuoteApi', 'api/SearchEquityDerivativeApi', 'api/StreamingDataApi', 'api/TradeBookApi', 'api/UserLimitsApi', 'api/UserLoginApi', 'api/UserLogoutApi'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('./ApiClient'), require('./model/AskValues'), require('./model/BidAsk'), require('./model/BidValues'), require('./model/CancelOrderResponse'), require('./model/CandleData'), require('./model/EquitySearchResponse'), require('./model/HistoricalCandleResponse'), require('./model/Holding'), require('./model/HoldingResponse'), require('./model/HoldingSummary'), require('./model/IndexCandleData'), require('./model/IndexCandleDataResponse'), require('./model/IndexIntraDayCandleData'), require('./model/IndexIntraDayCandleDataResponse'), require('./model/IndexQuoteResponse'), require('./model/IntraDayCandleData'), require('./model/IntraDayCandleResponses'), require('./model/Limit'), require('./model/LimitResponse'), require('./model/LoginRequest'), require('./model/LoginResponse'), require('./model/MarketDepthResponse'), require('./model/ModifyOrderRequest'), require('./model/OptionChain'), require('./model/OptionChainResponse'), require('./model/OrderBookEntry'), require('./model/OrderBookResponse'), require('./model/OrderDetails'), require('./model/OrderDetailsBO'), require('./model/OrderDetailsCO'), require('./model/OrderRequest'), require('./model/OrderRequestBO'), require('./model/OrderRequestCO'), require('./model/OrderResponse'), require('./model/OrderResponseBO'), require('./model/OrderResponseCO'), require('./model/OrderStatusResponse'), require('./model/Position'), require('./model/PositionConversionRequest'), require('./model/PositionConversionResponse'), require('./model/PositionResponse'), require('./model/PositionSquareOffListRequest'), require('./model/PositionSquareOffListResponse'), require('./model/PositionSquareOffRequest'), require('./model/PositionSummary'), require('./model/ScripDetails'), require('./model/SimpleResponse'), require('./model/StreamingMarkDataRequest'), require('./model/StreamingMarkDataResponse'), require('./model/StreamingQuoteDataRequest'), require('./model/StreamingQuoteDataResponse'), require('./model/TradeBookEntry'), require('./model/TradeBookResponse'), require('./model/TriggerOrder'), require('./model/TriggerOrdersResponse'), require('./api/HistoricalCandleDataApi'), require('./api/HoldingsApi'), require('./api/IntraDayCandleDataApi'), require('./api/OptionApi'), require('./api/OrdersApi'), require('./api/PositionsApi'), require('./api/QuoteApi'), require('./api/SearchEquityDerivativeApi'), require('./api/StreamingDataApi'), require('./api/TradeBookApi'), require('./api/UserLimitsApi'), require('./api/UserLoginApi'), require('./api/UserLogoutApi'));
  }
}(function(ApiClient, AskValues, BidAsk, BidValues, CancelOrderResponse, CandleData, EquitySearchResponse, HistoricalCandleResponse, Holding, HoldingResponse, HoldingSummary, IndexCandleData, IndexCandleDataResponse, IndexIntraDayCandleData, IndexIntraDayCandleDataResponse, IndexQuoteResponse, IntraDayCandleData, IntraDayCandleResponses, Limit, LimitResponse, LoginRequest, LoginResponse, MarketDepthResponse, ModifyOrderRequest, OptionChain, OptionChainResponse, OrderBookEntry, OrderBookResponse, OrderDetails, OrderDetailsBO, OrderDetailsCO, OrderRequest, OrderRequestBO, OrderRequestCO, OrderResponse, OrderResponseBO, OrderResponseCO, OrderStatusResponse, Position, PositionConversionRequest, PositionConversionResponse, PositionResponse, PositionSquareOffListRequest, PositionSquareOffListResponse, PositionSquareOffRequest, PositionSummary, ScripDetails, SimpleResponse, StreamingMarkDataRequest, StreamingMarkDataResponse, StreamingQuoteDataRequest, StreamingQuoteDataResponse, TradeBookEntry, TradeBookResponse, TriggerOrder, TriggerOrdersResponse, HistoricalCandleDataApi, HoldingsApi, IntraDayCandleDataApi, OptionApi, OrdersApi, PositionsApi, QuoteApi, SearchEquityDerivativeApi, StreamingDataApi, TradeBookApi, UserLimitsApi, UserLoginApi, UserLogoutApi) {
  'use strict';

  /**
   * _StockNote_API_is_a_set_of_Rest_APIs_using_which_users_can_build_customized_applications_based_on_their_trading_requirements__It_facilitates_the_users_of_the_APIs_to_login_search_symbols_place_orders_and_execute_them_view_their_order_status_positions_and_holdings_etc__This_documentation_provides_you_with_all_the_necessary_details_to_understand_the_SAMCO_StockNote_API_collection__APIs_are_compatible_with_both_Javascript_and_Python_For_any_issues_or_support_please_raise_ticket_to_us_using_the_a_hrefhttpswww_samco_insupportindextradeapisupport_linka_and_we_will_be_happy_to_assist_you___For_Reference_you_can_download_postman_collection_a_hrefhttpdevelopers_stocknote_comdocStockNoteApi_postman_collection_jsonClick_herea__For_downloading_a_list_of_all_tradeable_scrips_across_exchanges_please_a_hrefhttpsdevelopers_stocknote_comdocScripMaster_csvClick_herea_This_is_a_CSV_file_which_you_can_import_into_your_database__If_you_are_using_Python_or_Java_as_client_platform_to_code_your_strategies_and_integrate_with_our_APIs_you_can_use_our_StockNoteBridge__which_provides_pre_written_wrappers_over_StockNote_APIs_that_ensure_seamless_and_reliable_connectivity_to_the_APIs_and_help_you_fast_track_your_development_process_table_stylewidth100trthLanguagesththDocumentationthtrtrtdPythontdtda_hrefhttpsgithub_comsamco_sdkPython_SDKblobmasterREADME_mdPython_SDK_DocumentationatdtrtrtdJavatdtda_hrefhttpsgithub_comsamco_sdkJava_SDKblobmasterREADME_mdJava_SDK_Documentationatdtrtable_Add_session_token_of_login_api_response_as_x_session_token_in_header_param_of_all_Apis___NOTE_To_ensure_stability_and_there_by_provide_seamless_services_to_our_customers_Samco_may_set_limits_on_your_use_of_the_StockNote_APIs__for_example_limit_on_the_number_of_requests_sent_to_a_specific_API___If_you_have_additional_questions_regarding_the_rate_limits_on_APIs_please_reach_out_to_us_using_the_a_hrefhttpswww_samco_insupportindextradeapisupport_linka_and_we_will_be_happy_to_assist_you__.<br>
   * The <code>index</code> module provides access to constructors for all the classes which comprise the public API.
   * <p>
   * An AMD (recommended!) or CommonJS application will generally do something equivalent to the following:
   * <pre>
   * var StockNoteApiDocumentation = require('index'); // See note below*.
   * var xxxSvc = new StockNoteApiDocumentation.XxxApi(); // Allocate the API class we're going to use.
   * var yyyModel = new StockNoteApiDocumentation.Yyy(); // Construct a model instance.
   * yyyModel.someProperty = 'someValue';
   * ...
   * var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
   * ...
   * </pre>
   * <em>*NOTE: For a top-level AMD script, use require(['index'], function(){...})
   * and put the application logic within the callback function.</em>
   * </p>
   * <p>
   * A non-AMD browser application (discouraged) might do something like this:
   * <pre>
   * var xxxSvc = new StockNoteApiDocumentation.XxxApi(); // Allocate the API class we're going to use.
   * var yyy = new StockNoteApiDocumentation.Yyy(); // Construct a model instance.
   * yyyModel.someProperty = 'someValue';
   * ...
   * var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
   * ...
   * </pre>
   * </p>
   * @module index
   * @version 1.0
   */
  var exports = {
    /**
     * The ApiClient constructor.
     * @property {module:ApiClient}
     */
    ApiClient: ApiClient,
    /**
     * The AskValues model constructor.
     * @property {module:model/AskValues}
     */
    AskValues: AskValues,
    /**
     * The BidAsk model constructor.
     * @property {module:model/BidAsk}
     */
    BidAsk: BidAsk,
    /**
     * The BidValues model constructor.
     * @property {module:model/BidValues}
     */
    BidValues: BidValues,
    /**
     * The CancelOrderResponse model constructor.
     * @property {module:model/CancelOrderResponse}
     */
    CancelOrderResponse: CancelOrderResponse,
    /**
     * The CandleData model constructor.
     * @property {module:model/CandleData}
     */
    CandleData: CandleData,
    /**
     * The EquitySearchResponse model constructor.
     * @property {module:model/EquitySearchResponse}
     */
    EquitySearchResponse: EquitySearchResponse,
    /**
     * The HistoricalCandleResponse model constructor.
     * @property {module:model/HistoricalCandleResponse}
     */
    HistoricalCandleResponse: HistoricalCandleResponse,
    /**
     * The Holding model constructor.
     * @property {module:model/Holding}
     */
    Holding: Holding,
    /**
     * The HoldingResponse model constructor.
     * @property {module:model/HoldingResponse}
     */
    HoldingResponse: HoldingResponse,
    /**
     * The HoldingSummary model constructor.
     * @property {module:model/HoldingSummary}
     */
    HoldingSummary: HoldingSummary,
    /**
     * The IndexCandleData model constructor.
     * @property {module:model/IndexCandleData}
     */
    IndexCandleData: IndexCandleData,
    /**
     * The IndexCandleDataResponse model constructor.
     * @property {module:model/IndexCandleDataResponse}
     */
    IndexCandleDataResponse: IndexCandleDataResponse,
    /**
     * The IndexIntraDayCandleData model constructor.
     * @property {module:model/IndexIntraDayCandleData}
     */
    IndexIntraDayCandleData: IndexIntraDayCandleData,
    /**
     * The IndexIntraDayCandleDataResponse model constructor.
     * @property {module:model/IndexIntraDayCandleDataResponse}
     */
    IndexIntraDayCandleDataResponse: IndexIntraDayCandleDataResponse,
    /**
     * The IndexQuoteResponse model constructor.
     * @property {module:model/IndexQuoteResponse}
     */
    IndexQuoteResponse: IndexQuoteResponse,
    /**
     * The IntraDayCandleData model constructor.
     * @property {module:model/IntraDayCandleData}
     */
    IntraDayCandleData: IntraDayCandleData,
    /**
     * The IntraDayCandleResponses model constructor.
     * @property {module:model/IntraDayCandleResponses}
     */
    IntraDayCandleResponses: IntraDayCandleResponses,
    /**
     * The Limit model constructor.
     * @property {module:model/Limit}
     */
    Limit: Limit,
    /**
     * The LimitResponse model constructor.
     * @property {module:model/LimitResponse}
     */
    LimitResponse: LimitResponse,
    /**
     * The LoginRequest model constructor.
     * @property {module:model/LoginRequest}
     */
    LoginRequest: LoginRequest,
    /**
     * The LoginResponse model constructor.
     * @property {module:model/LoginResponse}
     */
    LoginResponse: LoginResponse,
    /**
     * The MarketDepthResponse model constructor.
     * @property {module:model/MarketDepthResponse}
     */
    MarketDepthResponse: MarketDepthResponse,
    /**
     * The ModifyOrderRequest model constructor.
     * @property {module:model/ModifyOrderRequest}
     */
    ModifyOrderRequest: ModifyOrderRequest,
    /**
     * The OptionChain model constructor.
     * @property {module:model/OptionChain}
     */
    OptionChain: OptionChain,
    /**
     * The OptionChainResponse model constructor.
     * @property {module:model/OptionChainResponse}
     */
    OptionChainResponse: OptionChainResponse,
    /**
     * The OrderBookEntry model constructor.
     * @property {module:model/OrderBookEntry}
     */
    OrderBookEntry: OrderBookEntry,
    /**
     * The OrderBookResponse model constructor.
     * @property {module:model/OrderBookResponse}
     */
    OrderBookResponse: OrderBookResponse,
    /**
     * The OrderDetails model constructor.
     * @property {module:model/OrderDetails}
     */
    OrderDetails: OrderDetails,
    /**
     * The OrderDetailsBO model constructor.
     * @property {module:model/OrderDetailsBO}
     */
    OrderDetailsBO: OrderDetailsBO,
    /**
     * The OrderDetailsCO model constructor.
     * @property {module:model/OrderDetailsCO}
     */
    OrderDetailsCO: OrderDetailsCO,
    /**
     * The OrderRequest model constructor.
     * @property {module:model/OrderRequest}
     */
    OrderRequest: OrderRequest,
    /**
     * The OrderRequestBO model constructor.
     * @property {module:model/OrderRequestBO}
     */
    OrderRequestBO: OrderRequestBO,
    /**
     * The OrderRequestCO model constructor.
     * @property {module:model/OrderRequestCO}
     */
    OrderRequestCO: OrderRequestCO,
    /**
     * The OrderResponse model constructor.
     * @property {module:model/OrderResponse}
     */
    OrderResponse: OrderResponse,
    /**
     * The OrderResponseBO model constructor.
     * @property {module:model/OrderResponseBO}
     */
    OrderResponseBO: OrderResponseBO,
    /**
     * The OrderResponseCO model constructor.
     * @property {module:model/OrderResponseCO}
     */
    OrderResponseCO: OrderResponseCO,
    /**
     * The OrderStatusResponse model constructor.
     * @property {module:model/OrderStatusResponse}
     */
    OrderStatusResponse: OrderStatusResponse,
    /**
     * The Position model constructor.
     * @property {module:model/Position}
     */
    Position: Position,
    /**
     * The PositionConversionRequest model constructor.
     * @property {module:model/PositionConversionRequest}
     */
    PositionConversionRequest: PositionConversionRequest,
    /**
     * The PositionConversionResponse model constructor.
     * @property {module:model/PositionConversionResponse}
     */
    PositionConversionResponse: PositionConversionResponse,
    /**
     * The PositionResponse model constructor.
     * @property {module:model/PositionResponse}
     */
    PositionResponse: PositionResponse,
    /**
     * The PositionSquareOffListRequest model constructor.
     * @property {module:model/PositionSquareOffListRequest}
     */
    PositionSquareOffListRequest: PositionSquareOffListRequest,
    /**
     * The PositionSquareOffListResponse model constructor.
     * @property {module:model/PositionSquareOffListResponse}
     */
    PositionSquareOffListResponse: PositionSquareOffListResponse,
    /**
     * The PositionSquareOffRequest model constructor.
     * @property {module:model/PositionSquareOffRequest}
     */
    PositionSquareOffRequest: PositionSquareOffRequest,
    /**
     * The PositionSummary model constructor.
     * @property {module:model/PositionSummary}
     */
    PositionSummary: PositionSummary,
    /**
     * The ScripDetails model constructor.
     * @property {module:model/ScripDetails}
     */
    ScripDetails: ScripDetails,
    /**
     * The SimpleResponse model constructor.
     * @property {module:model/SimpleResponse}
     */
    SimpleResponse: SimpleResponse,
    /**
     * The StreamingMarkDataRequest model constructor.
     * @property {module:model/StreamingMarkDataRequest}
     */
    StreamingMarkDataRequest: StreamingMarkDataRequest,
    /**
     * The StreamingMarkDataResponse model constructor.
     * @property {module:model/StreamingMarkDataResponse}
     */
    StreamingMarkDataResponse: StreamingMarkDataResponse,
    /**
     * The StreamingQuoteDataRequest model constructor.
     * @property {module:model/StreamingQuoteDataRequest}
     */
    StreamingQuoteDataRequest: StreamingQuoteDataRequest,
    /**
     * The StreamingQuoteDataResponse model constructor.
     * @property {module:model/StreamingQuoteDataResponse}
     */
    StreamingQuoteDataResponse: StreamingQuoteDataResponse,
    /**
     * The TradeBookEntry model constructor.
     * @property {module:model/TradeBookEntry}
     */
    TradeBookEntry: TradeBookEntry,
    /**
     * The TradeBookResponse model constructor.
     * @property {module:model/TradeBookResponse}
     */
    TradeBookResponse: TradeBookResponse,
    /**
     * The TriggerOrder model constructor.
     * @property {module:model/TriggerOrder}
     */
    TriggerOrder: TriggerOrder,
    /**
     * The TriggerOrdersResponse model constructor.
     * @property {module:model/TriggerOrdersResponse}
     */
    TriggerOrdersResponse: TriggerOrdersResponse,
    /**
     * The HistoricalCandleDataApi service constructor.
     * @property {module:api/HistoricalCandleDataApi}
     */
    HistoricalCandleDataApi: HistoricalCandleDataApi,
    /**
     * The HoldingsApi service constructor.
     * @property {module:api/HoldingsApi}
     */
    HoldingsApi: HoldingsApi,
    /**
     * The IntraDayCandleDataApi service constructor.
     * @property {module:api/IntraDayCandleDataApi}
     */
    IntraDayCandleDataApi: IntraDayCandleDataApi,
    /**
     * The OptionApi service constructor.
     * @property {module:api/OptionApi}
     */
    OptionApi: OptionApi,
    /**
     * The OrdersApi service constructor.
     * @property {module:api/OrdersApi}
     */
    OrdersApi: OrdersApi,
    /**
     * The PositionsApi service constructor.
     * @property {module:api/PositionsApi}
     */
    PositionsApi: PositionsApi,
    /**
     * The QuoteApi service constructor.
     * @property {module:api/QuoteApi}
     */
    QuoteApi: QuoteApi,
    /**
     * The SearchEquityDerivativeApi service constructor.
     * @property {module:api/SearchEquityDerivativeApi}
     */
    SearchEquityDerivativeApi: SearchEquityDerivativeApi,
    /**
     * The StreamingDataApi service constructor.
     * @property {module:api/StreamingDataApi}
     */
    StreamingDataApi: StreamingDataApi,
    /**
     * The TradeBookApi service constructor.
     * @property {module:api/TradeBookApi}
     */
    TradeBookApi: TradeBookApi,
    /**
     * The UserLimitsApi service constructor.
     * @property {module:api/UserLimitsApi}
     */
    UserLimitsApi: UserLimitsApi,
    /**
     * The UserLoginApi service constructor.
     * @property {module:api/UserLoginApi}
     */
    UserLoginApi: UserLoginApi,
    /**
     * The UserLogoutApi service constructor.
     * @property {module:api/UserLogoutApi}
     */
    UserLogoutApi: UserLogoutApi
  };

  return exports;
}));
