// stock_note_api_documentation/src/api/PositionsApi.js
/*
 * StockNote API Documentation
 *  StockNote API is a set of Rest APIs using which users can build customized applications based on their trading requirements. It facilitates the users of the APIs to login, search symbols, place orders and execute them, view their order status, positions and holdings etc. This documentation provides you with all the necessary details to understand the SAMCO StockNote API collection. APIs are compatible with both Javascript and Python.    For any issues or support, please raise ticket to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.      For Reference you can download postman collection <a href=\"http://developers.stocknote.com/doc/StockNoteApi.postman_collection.json\">Click here</a>      For downloading a list of all tradeable scrips across exchanges please <a href=\"https://developers.stocknote.com/doc/ScripMaster.csv\">Click here</a> This is a CSV file which you can import into your database.     If you are using Python or Java as client platform to code your strategies and integrate with our APIs, you can use our StockNoteBridge  which provides pre-written wrappers over StockNote APIs that ensure seamless and reliable connectivity to the APIs and help you fast-track your development process.  <table style=\"width:100%\"><tr><th>Languages</th><th>Documentation</th></tr><tr><td>Python</td><td><a href=\"https://github.com/samco-sdk/Python-SDK/blob/master/README.md\">Python SDK Documentation</a></td></tr><tr><td>Java</td><td><a href=\"https://github.com/samco-sdk/Java-SDK/blob/master/README.md\">Java SDK Documentation</a></td></tr></table> Add 'session token' of login api response as 'x-session-token' in header param of all Apis.      NOTE: To ensure stability and there by provide seamless services to our customers, Samco may set limits on your use of the StockNote APIs (for example, limit on the number of requests sent to a specific API) . If you have additional questions regarding the rate limits on APIs, please reach out to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.   
 *
 * OpenAPI spec version: 1.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.16
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/PositionConversionRequest', 'model/PositionConversionResponse', 'model/PositionResponse', 'model/PositionSquareOffListRequest', 'model/PositionSquareOffListResponse'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('../model/PositionConversionRequest'), require('../model/PositionConversionResponse'), require('../model/PositionResponse'), require('../model/PositionSquareOffListRequest'), require('../model/PositionSquareOffListResponse'));
  } else {
    // Browser globals (root is window)
    if (!root.StockNoteApiDocumentation) {
      root.StockNoteApiDocumentation = {};
    }
    root.StockNoteApiDocumentation.PositionsApi = factory(root.StockNoteApiDocumentation.ApiClient, root.StockNoteApiDocumentation.PositionConversionRequest, root.StockNoteApiDocumentation.PositionConversionResponse, root.StockNoteApiDocumentation.PositionResponse, root.StockNoteApiDocumentation.PositionSquareOffListRequest, root.StockNoteApiDocumentation.PositionSquareOffListResponse);
  }
}(this, function(ApiClient, PositionConversionRequest, PositionConversionResponse, PositionResponse, PositionSquareOffListRequest, PositionSquareOffListResponse) {
  'use strict';

  /**
   * Positions service.
   * @module api/PositionsApi
   * @version 1.0
   */

  /**
   * Constructs a new PositionsApi. 
   * @alias module:api/PositionsApi
   * @class
   * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
   * default to {@link module:ApiClient#instance} if unspecified.
   */
  var exports = function(apiClient) {
    this.apiClient = apiClient || ApiClient.instance;


    /**
     * Callback function to receive the result of the convertPosition operation.
     * @callback module:api/PositionsApi~convertPositionCallback
     * @param {String} error Error message, if any.
     * @param {module:model/PositionConversionResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Position Conversion
     * Convert an existing position of a margin product to a different margin product type. All or a subset of an existing position quantity can be converted to a different product type.The available margin product types are MARGIN_INTRADAY_SQUAREOFF(MIS), CASHNCARRY(CNC), NORMAL(NRML).
     * @param {String} xSessionToken Session token that should be used with API request is given in the response. It is used as an authenticator. A session token is valid for 24 hours or until a new login request is given, where you a new session token will be generated and the previous one will get expired.
     * @param {Object} opts Optional parameters
     * @param {module:model/PositionConversionRequest} opts.body 
     * @param {module:api/PositionsApi~convertPositionCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/PositionConversionResponse}
     */
    this.convertPosition = function(xSessionToken, opts, callback) {
      opts = opts || {};
      var postBody = opts['body'];

      // verify the required parameter 'xSessionToken' is set
      if (xSessionToken === undefined || xSessionToken === null) {
        throw new Error("Missing the required parameter 'xSessionToken' when calling convertPosition");
      }


      var pathParams = {
      };
      var queryParams = {
      };
      var collectionQueryParams = {
      };
      var headerParams = {
        'x-session-token': xSessionToken
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = PositionConversionResponse;

      return this.apiClient.callApi(
        '/position/convertPosition', 'POST',
        pathParams, queryParams, collectionQueryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }

    /**
     * Callback function to receive the result of the getPositions operation.
     * @callback module:api/PositionsApi~getPositionsCallback
     * @param {String} error Error message, if any.
     * @param {module:model/PositionResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * User Positions
     * Get position details of the user (The details of equity, derivative, commodity, currency borrowed or owned by the user).
     * @param {String} xSessionToken Session token that should be used with API request is given in the response. It is used as an authenticator. A session token is valid for 24 hours or until a new login request is given, where you a new session token will be generated and the previous one will get expired.
     * @param {String} positionType Position type can be DAY/NET. If Position type is “DAY”, fetch current day position details, If Position Type is “NET”, fetch carry forward position details
     * @param {module:api/PositionsApi~getPositionsCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/PositionResponse}
     */
    this.getPositions = function(xSessionToken, positionType, callback) {
      var postBody = null;

      // verify the required parameter 'xSessionToken' is set
      if (xSessionToken === undefined || xSessionToken === null) {
        throw new Error("Missing the required parameter 'xSessionToken' when calling getPositions");
      }

      // verify the required parameter 'positionType' is set
      if (positionType === undefined || positionType === null) {
        throw new Error("Missing the required parameter 'positionType' when calling getPositions");
      }


      var pathParams = {
      };
      var queryParams = {
        'positionType': positionType,
      };
      var collectionQueryParams = {
      };
      var headerParams = {
        'x-session-token': xSessionToken
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = PositionResponse;

      return this.apiClient.callApi(
        '/position/getPositions', 'GET',
        pathParams, queryParams, collectionQueryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }

    /**
     * Callback function to receive the result of the squareOffPosition operation.
     * @callback module:api/PositionsApi~squareOffPositionCallback
     * @param {String} error Error message, if any.
     * @param {module:model/PositionSquareOffListResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Position Square Off
     * SqareOff existing position. Mostly used in day trading, in which user buy or sell a particular quantity of a stock and later in the day reverse the transaction to earn a profit.
     * @param {String} xSessionToken Session token that should be used with API request is given in the response. It is used as an authenticator. A session token is valid for 24 hours or until a new login request is given, where you a new session token will be generated and the previous one will get expired.
     * @param {Object} opts Optional parameters
     * @param {module:model/PositionSquareOffListRequest} opts.body 
     * @param {module:api/PositionsApi~squareOffPositionCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/PositionSquareOffListResponse}
     */
    this.squareOffPosition = function(xSessionToken, opts, callback) {
      opts = opts || {};
      var postBody = opts['body'];

      // verify the required parameter 'xSessionToken' is set
      if (xSessionToken === undefined || xSessionToken === null) {
        throw new Error("Missing the required parameter 'xSessionToken' when calling squareOffPosition");
      }


      var pathParams = {
      };
      var queryParams = {
      };
      var collectionQueryParams = {
      };
      var headerParams = {
        'x-session-token': xSessionToken
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = PositionSquareOffListResponse;

      return this.apiClient.callApi(
        '/position/squareOff', 'POST',
        pathParams, queryParams, collectionQueryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }
  };

  return exports;
}));
