// stock_note_api_documentation/src/model/HoldingSummary.js
/*
 * StockNote API Documentation
 *  StockNote API is a set of Rest APIs using which users can build customized applications based on their trading requirements. It facilitates the users of the APIs to login, search symbols, place orders and execute them, view their order status, positions and holdings etc. This documentation provides you with all the necessary details to understand the SAMCO StockNote API collection. APIs are compatible with both Javascript and Python.    For any issues or support, please raise ticket to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.      For Reference you can download postman collection <a href=\"http://developers.stocknote.com/doc/StockNoteApi.postman_collection.json\">Click here</a>      For downloading a list of all tradeable scrips across exchanges please <a href=\"https://developers.stocknote.com/doc/ScripMaster.csv\">Click here</a> This is a CSV file which you can import into your database.     If you are using Python or Java as client platform to code your strategies and integrate with our APIs, you can use our StockNoteBridge  which provides pre-written wrappers over StockNote APIs that ensure seamless and reliable connectivity to the APIs and help you fast-track your development process.  <table style=\"width:100%\"><tr><th>Languages</th><th>Documentation</th></tr><tr><td>Python</td><td><a href=\"https://github.com/samco-sdk/Python-SDK/blob/master/README.md\">Python SDK Documentation</a></td></tr><tr><td>Java</td><td><a href=\"https://github.com/samco-sdk/Java-SDK/blob/master/README.md\">Java SDK Documentation</a></td></tr></table> Add 'session token' of login api response as 'x-session-token' in header param of all Apis.      NOTE: To ensure stability and there by provide seamless services to our customers, Samco may set limits on your use of the StockNote APIs (for example, limit on the number of requests sent to a specific API) . If you have additional questions regarding the rate limits on APIs, please reach out to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.   
 *
 * OpenAPI spec version: 1.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.16
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.StockNoteApiDocumentation) {
      root.StockNoteApiDocumentation = {};
    }
    root.StockNoteApiDocumentation.HoldingSummary = factory(root.StockNoteApiDocumentation.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The HoldingSummary model module.
   * @module model/HoldingSummary
   * @version 1.0
   */

  /**
   * Constructs a new <code>HoldingSummary</code>.
   * @alias module:model/HoldingSummary
   * @class
   * @param gainingTodayCount {String} Count of scrips which are gaining in value in a trading day
   * @param losingTodayCount {String} Count of scrips which are losing in value in a trading day
   * @param totalGainAndLossAmount {String} Total amount of Gain / Loss for all existing positions since their creation
   * @param dayGainAndLossAmount {String} Amount of Gain / Loss for all positions on a specific trading day
   * @param portfolioValue {String} Value of the portfolio.
   */
  var exports = function(gainingTodayCount, losingTodayCount, totalGainAndLossAmount, dayGainAndLossAmount, portfolioValue) {
    this.gainingTodayCount = gainingTodayCount;
    this.losingTodayCount = losingTodayCount;
    this.totalGainAndLossAmount = totalGainAndLossAmount;
    this.dayGainAndLossAmount = dayGainAndLossAmount;
    this.portfolioValue = portfolioValue;
  };

  /**
   * Constructs a <code>HoldingSummary</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/HoldingSummary} obj Optional instance to populate.
   * @return {module:model/HoldingSummary} The populated <code>HoldingSummary</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();
      if (data.hasOwnProperty('gainingTodayCount'))
        obj.gainingTodayCount = ApiClient.convertToType(data['gainingTodayCount'], 'String');
      if (data.hasOwnProperty('losingTodayCount'))
        obj.losingTodayCount = ApiClient.convertToType(data['losingTodayCount'], 'String');
      if (data.hasOwnProperty('totalGainAndLossAmount'))
        obj.totalGainAndLossAmount = ApiClient.convertToType(data['totalGainAndLossAmount'], 'String');
      if (data.hasOwnProperty('dayGainAndLossAmount'))
        obj.dayGainAndLossAmount = ApiClient.convertToType(data['dayGainAndLossAmount'], 'String');
      if (data.hasOwnProperty('portfolioValue'))
        obj.portfolioValue = ApiClient.convertToType(data['portfolioValue'], 'String');
    }
    return obj;
  }

  /**
   * Count of scrips which are gaining in value in a trading day
   * @member {String} gainingTodayCount
   */
  exports.prototype.gainingTodayCount = undefined;

  /**
   * Count of scrips which are losing in value in a trading day
   * @member {String} losingTodayCount
   */
  exports.prototype.losingTodayCount = undefined;

  /**
   * Total amount of Gain / Loss for all existing positions since their creation
   * @member {String} totalGainAndLossAmount
   */
  exports.prototype.totalGainAndLossAmount = undefined;

  /**
   * Amount of Gain / Loss for all positions on a specific trading day
   * @member {String} dayGainAndLossAmount
   */
  exports.prototype.dayGainAndLossAmount = undefined;

  /**
   * Value of the portfolio.
   * @member {String} portfolioValue
   */
  exports.prototype.portfolioValue = undefined;

  return exports;

}));
