// stock_note_api_documentation/src/model/IndexQuoteResponse.js
/*
 * StockNote API Documentation
 *  StockNote API is a set of Rest APIs using which users can build customized applications based on their trading requirements. It facilitates the users of the APIs to login, search symbols, place orders and execute them, view their order status, positions and holdings etc. This documentation provides you with all the necessary details to understand the SAMCO StockNote API collection. APIs are compatible with both Javascript and Python.    For any issues or support, please raise ticket to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.      For Reference you can download postman collection <a href=\"http://developers.stocknote.com/doc/StockNoteApi.postman_collection.json\">Click here</a>      For downloading a list of all tradeable scrips across exchanges please <a href=\"https://developers.stocknote.com/doc/ScripMaster.csv\">Click here</a> This is a CSV file which you can import into your database.     If you are using Python or Java as client platform to code your strategies and integrate with our APIs, you can use our StockNoteBridge  which provides pre-written wrappers over StockNote APIs that ensure seamless and reliable connectivity to the APIs and help you fast-track your development process.  <table style=\"width:100%\"><tr><th>Languages</th><th>Documentation</th></tr><tr><td>Python</td><td><a href=\"https://github.com/samco-sdk/Python-SDK/blob/master/README.md\">Python SDK Documentation</a></td></tr><tr><td>Java</td><td><a href=\"https://github.com/samco-sdk/Java-SDK/blob/master/README.md\">Java SDK Documentation</a></td></tr></table> Add 'session token' of login api response as 'x-session-token' in header param of all Apis.      NOTE: To ensure stability and there by provide seamless services to our customers, Samco may set limits on your use of the StockNote APIs (for example, limit on the number of requests sent to a specific API) . If you have additional questions regarding the rate limits on APIs, please reach out to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.   
 *
 * OpenAPI spec version: 1.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.16
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.StockNoteApiDocumentation) {
      root.StockNoteApiDocumentation = {};
    }
    root.StockNoteApiDocumentation.IndexQuoteResponse = factory(root.StockNoteApiDocumentation.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The IndexQuoteResponse model module.
   * @module model/IndexQuoteResponse
   * @version 1.0
   */

  /**
   * Constructs a new <code>IndexQuoteResponse</code>.
   * @alias module:model/IndexQuoteResponse
   * @class
   * @param serverTime {String} Time at Server.
   * @param msgId {String} This is a unique identifier for every request into the system. Please quote this identifier to the support team if you face issues with the API request.
   * @param status {String} Response status. Can be Success or Failure
   * @param statusMessage {String} Status message of the Quote request
   * @param indexName {String} Index Name of the indices.
   * @param listingId {String} Identifier assigned to the scrip by exchange in the format <<listing code>>_<<exchange>>
   * @param lastTradedTime {String} Time of the last transaction
   * @param lastTradedPrice {String} Price at which last transaction / trade is done
   * @param spotPrice {String} Spot price. Applicable in case of Futures and Options
   * @param changePercentage {String} Percentage of change between the current value and the previous day's market close
   * @param lastTradedQuantity {String} Quantity of last transaction
   * @param averagePrice {String} average price of a market snapshot
   * @param openValue {String} Opening price of a market snapshot
   * @param highValue {String} High value of market snapshot
   * @param lowValue {String} Low value of market snapshot
   * @param closeValue {String} Close value of market snapshot
   * @param totalBuyQuantity {String} Total quantity of BUY transaction
   * @param totalSellQuantity {String} Total quantity of SELL transaction
   * @param totalTradedValue {String} Value of total trade made for the scrip
   * @param totalTradedVolume {String} Total volume of trading done
   * @param openInterest {String} Open interest is the Number of existing contracts held by buyers or sellers for any market for any given day.
   * @param oIChangePer {String} It will shows the change % based on Open Interest
   */
  var exports = function(serverTime, msgId, status, statusMessage, indexName, listingId, lastTradedTime, lastTradedPrice, spotPrice, changePercentage, lastTradedQuantity, averagePrice, openValue, highValue, lowValue, closeValue, totalBuyQuantity, totalSellQuantity, totalTradedValue, totalTradedVolume, openInterest, oIChangePer) {
    this.serverTime = serverTime;
    this.msgId = msgId;
    this.status = status;
    this.statusMessage = statusMessage;
    this.indexName = indexName;
    this.listingId = listingId;
    this.lastTradedTime = lastTradedTime;
    this.lastTradedPrice = lastTradedPrice;
    this.spotPrice = spotPrice;
    this.changePercentage = changePercentage;
    this.lastTradedQuantity = lastTradedQuantity;
    this.averagePrice = averagePrice;
    this.openValue = openValue;
    this.highValue = highValue;
    this.lowValue = lowValue;
    this.closeValue = closeValue;
    this.totalBuyQuantity = totalBuyQuantity;
    this.totalSellQuantity = totalSellQuantity;
    this.totalTradedValue = totalTradedValue;
    this.totalTradedVolume = totalTradedVolume;
    this.openInterest = openInterest;
    this.oIChangePer = oIChangePer;
  };

  /**
   * Constructs a <code>IndexQuoteResponse</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/IndexQuoteResponse} obj Optional instance to populate.
   * @return {module:model/IndexQuoteResponse} The populated <code>IndexQuoteResponse</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();
      if (data.hasOwnProperty('serverTime'))
        obj.serverTime = ApiClient.convertToType(data['serverTime'], 'String');
      if (data.hasOwnProperty('msgId'))
        obj.msgId = ApiClient.convertToType(data['msgId'], 'String');
      if (data.hasOwnProperty('status'))
        obj.status = ApiClient.convertToType(data['status'], 'String');
      if (data.hasOwnProperty('statusMessage'))
        obj.statusMessage = ApiClient.convertToType(data['statusMessage'], 'String');
      if (data.hasOwnProperty('indexName'))
        obj.indexName = ApiClient.convertToType(data['indexName'], 'String');
      if (data.hasOwnProperty('listingId'))
        obj.listingId = ApiClient.convertToType(data['listingId'], 'String');
      if (data.hasOwnProperty('lastTradedTime'))
        obj.lastTradedTime = ApiClient.convertToType(data['lastTradedTime'], 'String');
      if (data.hasOwnProperty('lastTradedPrice'))
        obj.lastTradedPrice = ApiClient.convertToType(data['lastTradedPrice'], 'String');
      if (data.hasOwnProperty('spotPrice'))
        obj.spotPrice = ApiClient.convertToType(data['spotPrice'], 'String');
      if (data.hasOwnProperty('changePercentage'))
        obj.changePercentage = ApiClient.convertToType(data['changePercentage'], 'String');
      if (data.hasOwnProperty('lastTradedQuantity'))
        obj.lastTradedQuantity = ApiClient.convertToType(data['lastTradedQuantity'], 'String');
      if (data.hasOwnProperty('averagePrice'))
        obj.averagePrice = ApiClient.convertToType(data['averagePrice'], 'String');
      if (data.hasOwnProperty('openValue'))
        obj.openValue = ApiClient.convertToType(data['openValue'], 'String');
      if (data.hasOwnProperty('highValue'))
        obj.highValue = ApiClient.convertToType(data['highValue'], 'String');
      if (data.hasOwnProperty('lowValue'))
        obj.lowValue = ApiClient.convertToType(data['lowValue'], 'String');
      if (data.hasOwnProperty('closeValue'))
        obj.closeValue = ApiClient.convertToType(data['closeValue'], 'String');
      if (data.hasOwnProperty('totalBuyQuantity'))
        obj.totalBuyQuantity = ApiClient.convertToType(data['totalBuyQuantity'], 'String');
      if (data.hasOwnProperty('totalSellQuantity'))
        obj.totalSellQuantity = ApiClient.convertToType(data['totalSellQuantity'], 'String');
      if (data.hasOwnProperty('totalTradedValue'))
        obj.totalTradedValue = ApiClient.convertToType(data['totalTradedValue'], 'String');
      if (data.hasOwnProperty('totalTradedVolume'))
        obj.totalTradedVolume = ApiClient.convertToType(data['totalTradedVolume'], 'String');
      if (data.hasOwnProperty('openInterest'))
        obj.openInterest = ApiClient.convertToType(data['openInterest'], 'String');
      if (data.hasOwnProperty('oIChangePer'))
        obj.oIChangePer = ApiClient.convertToType(data['oIChangePer'], 'String');
    }
    return obj;
  }

  /**
   * Time at Server.
   * @member {String} serverTime
   */
  exports.prototype.serverTime = undefined;

  /**
   * This is a unique identifier for every request into the system. Please quote this identifier to the support team if you face issues with the API request.
   * @member {String} msgId
   */
  exports.prototype.msgId = undefined;

  /**
   * Response status. Can be Success or Failure
   * @member {String} status
   */
  exports.prototype.status = undefined;

  /**
   * Status message of the Quote request
   * @member {String} statusMessage
   */
  exports.prototype.statusMessage = undefined;

  /**
   * Index Name of the indices.
   * @member {String} indexName
   */
  exports.prototype.indexName = undefined;

  /**
   * Identifier assigned to the scrip by exchange in the format <<listing code>>_<<exchange>>
   * @member {String} listingId
   */
  exports.prototype.listingId = undefined;

  /**
   * Time of the last transaction
   * @member {String} lastTradedTime
   */
  exports.prototype.lastTradedTime = undefined;

  /**
   * Price at which last transaction / trade is done
   * @member {String} lastTradedPrice
   */
  exports.prototype.lastTradedPrice = undefined;

  /**
   * Spot price. Applicable in case of Futures and Options
   * @member {String} spotPrice
   */
  exports.prototype.spotPrice = undefined;

  /**
   * Percentage of change between the current value and the previous day's market close
   * @member {String} changePercentage
   */
  exports.prototype.changePercentage = undefined;

  /**
   * Quantity of last transaction
   * @member {String} lastTradedQuantity
   */
  exports.prototype.lastTradedQuantity = undefined;

  /**
   * average price of a market snapshot
   * @member {String} averagePrice
   */
  exports.prototype.averagePrice = undefined;

  /**
   * Opening price of a market snapshot
   * @member {String} openValue
   */
  exports.prototype.openValue = undefined;

  /**
   * High value of market snapshot
   * @member {String} highValue
   */
  exports.prototype.highValue = undefined;

  /**
   * Low value of market snapshot
   * @member {String} lowValue
   */
  exports.prototype.lowValue = undefined;

  /**
   * Close value of market snapshot
   * @member {String} closeValue
   */
  exports.prototype.closeValue = undefined;

  /**
   * Total quantity of BUY transaction
   * @member {String} totalBuyQuantity
   */
  exports.prototype.totalBuyQuantity = undefined;

  /**
   * Total quantity of SELL transaction
   * @member {String} totalSellQuantity
   */
  exports.prototype.totalSellQuantity = undefined;

  /**
   * Value of total trade made for the scrip
   * @member {String} totalTradedValue
   */
  exports.prototype.totalTradedValue = undefined;

  /**
   * Total volume of trading done
   * @member {String} totalTradedVolume
   */
  exports.prototype.totalTradedVolume = undefined;

  /**
   * Open interest is the Number of existing contracts held by buyers or sellers for any market for any given day.
   * @member {String} openInterest
   */
  exports.prototype.openInterest = undefined;

  /**
   * It will shows the change % based on Open Interest
   * @member {String} oIChangePer
   */
  exports.prototype.oIChangePer = undefined;

  return exports;

}));
