// stock_note_api_documentation/src/model/IndexIntraDayCandleData.js
/*
 * StockNote API Documentation
 *  StockNote API is a set of Rest APIs using which users can build customized applications based on their trading requirements. It facilitates the users of the APIs to login, search symbols, place orders and execute them, view their order status, positions and holdings etc. This documentation provides you with all the necessary details to understand the SAMCO StockNote API collection. APIs are compatible with both Javascript and Python.    For any issues or support, please raise ticket to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.      For Reference you can download postman collection <a href=\"http://developers.stocknote.com/doc/StockNoteApi.postman_collection.json\">Click here</a>      For downloading a list of all tradeable scrips across exchanges please <a href=\"https://developers.stocknote.com/doc/ScripMaster.csv\">Click here</a> This is a CSV file which you can import into your database.     If you are using Python or Java as client platform to code your strategies and integrate with our APIs, you can use our StockNoteBridge  which provides pre-written wrappers over StockNote APIs that ensure seamless and reliable connectivity to the APIs and help you fast-track your development process.  <table style=\"width:100%\"><tr><th>Languages</th><th>Documentation</th></tr><tr><td>Python</td><td><a href=\"https://github.com/samco-sdk/Python-SDK/blob/master/README.md\">Python SDK Documentation</a></td></tr><tr><td>Java</td><td><a href=\"https://github.com/samco-sdk/Java-SDK/blob/master/README.md\">Java SDK Documentation</a></td></tr></table> Add 'session token' of login api response as 'x-session-token' in header param of all Apis.      NOTE: To ensure stability and there by provide seamless services to our customers, Samco may set limits on your use of the StockNote APIs (for example, limit on the number of requests sent to a specific API) . If you have additional questions regarding the rate limits on APIs, please reach out to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.   
 *
 * OpenAPI spec version: 1.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.16
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.StockNoteApiDocumentation) {
      root.StockNoteApiDocumentation = {};
    }
    root.StockNoteApiDocumentation.IndexIntraDayCandleData = factory(root.StockNoteApiDocumentation.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The IndexIntraDayCandleData model module.
   * @module model/IndexIntraDayCandleData
   * @version 1.0
   */

  /**
   * Constructs a new <code>IndexIntraDayCandleData</code>.
   * @alias module:model/IndexIntraDayCandleData
   * @class
   * @param dateTime {String} Date for which CandleData shown
   * @param open {String} Opening price of a market snapshot
   * @param high {String} High value of market snapshot
   * @param low {String} Low value of market snapshot
   * @param close {String} Close value of market snapshot
   * @param volume {String} Limit amount of a security traded on the specific day
   */
  var exports = function(dateTime, open, high, low, close, volume) {
    this.dateTime = dateTime;
    this.open = open;
    this.high = high;
    this.low = low;
    this.close = close;
    this.volume = volume;
  };

  /**
   * Constructs a <code>IndexIntraDayCandleData</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/IndexIntraDayCandleData} obj Optional instance to populate.
   * @return {module:model/IndexIntraDayCandleData} The populated <code>IndexIntraDayCandleData</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();
      if (data.hasOwnProperty('dateTime'))
        obj.dateTime = ApiClient.convertToType(data['dateTime'], 'String');
      if (data.hasOwnProperty('open'))
        obj.open = ApiClient.convertToType(data['open'], 'String');
      if (data.hasOwnProperty('high'))
        obj.high = ApiClient.convertToType(data['high'], 'String');
      if (data.hasOwnProperty('low'))
        obj.low = ApiClient.convertToType(data['low'], 'String');
      if (data.hasOwnProperty('close'))
        obj.close = ApiClient.convertToType(data['close'], 'String');
      if (data.hasOwnProperty('volume'))
        obj.volume = ApiClient.convertToType(data['volume'], 'String');
    }
    return obj;
  }

  /**
   * Date for which CandleData shown
   * @member {String} dateTime
   */
  exports.prototype.dateTime = undefined;

  /**
   * Opening price of a market snapshot
   * @member {String} open
   */
  exports.prototype.open = undefined;

  /**
   * High value of market snapshot
   * @member {String} high
   */
  exports.prototype.high = undefined;

  /**
   * Low value of market snapshot
   * @member {String} low
   */
  exports.prototype.low = undefined;

  /**
   * Close value of market snapshot
   * @member {String} close
   */
  exports.prototype.close = undefined;

  /**
   * Limit amount of a security traded on the specific day
   * @member {String} volume
   */
  exports.prototype.volume = undefined;

  return exports;

}));
