// stock_note_api_documentation/src/model/Limit.js
/*
 * StockNote API Documentation
 *  StockNote API is a set of Rest APIs using which users can build customized applications based on their trading requirements. It facilitates the users of the APIs to login, search symbols, place orders and execute them, view their order status, positions and holdings etc. This documentation provides you with all the necessary details to understand the SAMCO StockNote API collection. APIs are compatible with both Javascript and Python.    For any issues or support, please raise ticket to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.      For Reference you can download postman collection <a href=\"http://developers.stocknote.com/doc/StockNoteApi.postman_collection.json\">Click here</a>      For downloading a list of all tradeable scrips across exchanges please <a href=\"https://developers.stocknote.com/doc/ScripMaster.csv\">Click here</a> This is a CSV file which you can import into your database.     If you are using Python or Java as client platform to code your strategies and integrate with our APIs, you can use our StockNoteBridge  which provides pre-written wrappers over StockNote APIs that ensure seamless and reliable connectivity to the APIs and help you fast-track your development process.  <table style=\"width:100%\"><tr><th>Languages</th><th>Documentation</th></tr><tr><td>Python</td><td><a href=\"https://github.com/samco-sdk/Python-SDK/blob/master/README.md\">Python SDK Documentation</a></td></tr><tr><td>Java</td><td><a href=\"https://github.com/samco-sdk/Java-SDK/blob/master/README.md\">Java SDK Documentation</a></td></tr></table> Add 'session token' of login api response as 'x-session-token' in header param of all Apis.      NOTE: To ensure stability and there by provide seamless services to our customers, Samco may set limits on your use of the StockNote APIs (for example, limit on the number of requests sent to a specific API) . If you have additional questions regarding the rate limits on APIs, please reach out to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.   
 *
 * OpenAPI spec version: 1.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.16
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.StockNoteApiDocumentation) {
      root.StockNoteApiDocumentation = {};
    }
    root.StockNoteApiDocumentation.Limit = factory(root.StockNoteApiDocumentation.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The Limit model module.
   * @module model/Limit
   * @version 1.0
   */

  /**
   * Constructs a new <code>Limit</code>.
   * @alias module:model/Limit
   * @class
   * @param grossAvailableMargin {String} Total amount of margins or balances available for trading. Opening balance for current day
   * @param payInToday {String} Current day deposited amount
   * @param notionalCash {String} Additional Limit that may or may not be given by RMS for Trading.
   * @param collateralMarginAgainstShares {String} Margin against shares offered by SAMCO to their clients for trading in stock and shares.
   * @param marginUsed {String} The amount deducted from opening balance for trading on the current day, and the amount blocked for creating a position when user places an order.
   * @param netAvailableMargin {String} Actual margin available with user for Trading after making all necessary adjustments.
   */
  var exports = function(grossAvailableMargin, payInToday, notionalCash, collateralMarginAgainstShares, marginUsed, netAvailableMargin) {
    this.grossAvailableMargin = grossAvailableMargin;
    this.payInToday = payInToday;
    this.notionalCash = notionalCash;
    this.collateralMarginAgainstShares = collateralMarginAgainstShares;
    this.marginUsed = marginUsed;
    this.netAvailableMargin = netAvailableMargin;
  };

  /**
   * Constructs a <code>Limit</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Limit} obj Optional instance to populate.
   * @return {module:model/Limit} The populated <code>Limit</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();
      if (data.hasOwnProperty('grossAvailableMargin'))
        obj.grossAvailableMargin = ApiClient.convertToType(data['grossAvailableMargin'], 'String');
      if (data.hasOwnProperty('payInToday'))
        obj.payInToday = ApiClient.convertToType(data['payInToday'], 'String');
      if (data.hasOwnProperty('notionalCash'))
        obj.notionalCash = ApiClient.convertToType(data['notionalCash'], 'String');
      if (data.hasOwnProperty('collateralMarginAgainstShares'))
        obj.collateralMarginAgainstShares = ApiClient.convertToType(data['collateralMarginAgainstShares'], 'String');
      if (data.hasOwnProperty('marginUsed'))
        obj.marginUsed = ApiClient.convertToType(data['marginUsed'], 'String');
      if (data.hasOwnProperty('netAvailableMargin'))
        obj.netAvailableMargin = ApiClient.convertToType(data['netAvailableMargin'], 'String');
    }
    return obj;
  }

  /**
   * Total amount of margins or balances available for trading. Opening balance for current day
   * @member {String} grossAvailableMargin
   */
  exports.prototype.grossAvailableMargin = undefined;

  /**
   * Current day deposited amount
   * @member {String} payInToday
   */
  exports.prototype.payInToday = undefined;

  /**
   * Additional Limit that may or may not be given by RMS for Trading.
   * @member {String} notionalCash
   */
  exports.prototype.notionalCash = undefined;

  /**
   * Margin against shares offered by SAMCO to their clients for trading in stock and shares.
   * @member {String} collateralMarginAgainstShares
   */
  exports.prototype.collateralMarginAgainstShares = undefined;

  /**
   * The amount deducted from opening balance for trading on the current day, and the amount blocked for creating a position when user places an order.
   * @member {String} marginUsed
   */
  exports.prototype.marginUsed = undefined;

  /**
   * Actual margin available with user for Trading after making all necessary adjustments.
   * @member {String} netAvailableMargin
   */
  exports.prototype.netAvailableMargin = undefined;

  return exports;

}));
