// stock_note_api_documentation/src/model/LoginResponse.js
/*
 * StockNote API Documentation
 *  StockNote API is a set of Rest APIs using which users can build customized applications based on their trading requirements. It facilitates the users of the APIs to login, search symbols, place orders and execute them, view their order status, positions and holdings etc. This documentation provides you with all the necessary details to understand the SAMCO StockNote API collection. APIs are compatible with both Javascript and Python.    For any issues or support, please raise ticket to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.      For Reference you can download postman collection <a href=\"http://developers.stocknote.com/doc/StockNoteApi.postman_collection.json\">Click here</a>      For downloading a list of all tradeable scrips across exchanges please <a href=\"https://developers.stocknote.com/doc/ScripMaster.csv\">Click here</a> This is a CSV file which you can import into your database.     If you are using Python or Java as client platform to code your strategies and integrate with our APIs, you can use our StockNoteBridge  which provides pre-written wrappers over StockNote APIs that ensure seamless and reliable connectivity to the APIs and help you fast-track your development process.  <table style=\"width:100%\"><tr><th>Languages</th><th>Documentation</th></tr><tr><td>Python</td><td><a href=\"https://github.com/samco-sdk/Python-SDK/blob/master/README.md\">Python SDK Documentation</a></td></tr><tr><td>Java</td><td><a href=\"https://github.com/samco-sdk/Java-SDK/blob/master/README.md\">Java SDK Documentation</a></td></tr></table> Add 'session token' of login api response as 'x-session-token' in header param of all Apis.      NOTE: To ensure stability and there by provide seamless services to our customers, Samco may set limits on your use of the StockNote APIs (for example, limit on the number of requests sent to a specific API) . If you have additional questions regarding the rate limits on APIs, please reach out to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.   
 *
 * OpenAPI spec version: 1.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.16
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.StockNoteApiDocumentation) {
      root.StockNoteApiDocumentation = {};
    }
    root.StockNoteApiDocumentation.LoginResponse = factory(root.StockNoteApiDocumentation.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The LoginResponse model module.
   * @module model/LoginResponse
   * @version 1.0
   */

  /**
   * Constructs a new <code>LoginResponse</code>.
   * @alias module:model/LoginResponse
   * @class
   * @param serverTime {String} Time at Server.
   * @param msgId {String} This is a unique identifier for every request into the system. Please quote this identifier to the support team if you face issues with the API request.
   * @param status {String} Login Success or Failure.
   * @param statusMessage {String} status Message
   * @param sessionToken {String} Session token that should be used with API request is given in the response. It is used as an authenticator. A session token is valid for 24 hours or until a new login request is given, where you a new session token will be generated and the previous one will get expired.
   * @param accountID {String} Same as Client Code for SAMCO Trading Account, is unique and used for identification of user.
   * @param accountName {String} Users full name as registered with Samco Trading Account.
   * @param exchangeList {Array.<String>} List of exchanges enabled for trading for the user.
   * @param orderTypeList {Array.<String>} List of order types enabled for trading for the user.The list can be from of the following, MKT - Market Order,L- Limit Order,SL - Stop Loss Limit,SL-M - Stop loss market.
   * @param productList {Array.<String>} List of product types enabled for the user . It can be CNC (Cash and Carry),BO (Bracket Order),CO (Cover Order),NRML (Normal),MIS (Intraday).
   */
  var exports = function(serverTime, msgId, status, statusMessage, sessionToken, accountID, accountName, exchangeList, orderTypeList, productList) {
    this.serverTime = serverTime;
    this.msgId = msgId;
    this.status = status;
    this.statusMessage = statusMessage;
    this.sessionToken = sessionToken;
    this.accountID = accountID;
    this.accountName = accountName;
    this.exchangeList = exchangeList;
    this.orderTypeList = orderTypeList;
    this.productList = productList;
  };

  /**
   * Constructs a <code>LoginResponse</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/LoginResponse} obj Optional instance to populate.
   * @return {module:model/LoginResponse} The populated <code>LoginResponse</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();
      if (data.hasOwnProperty('serverTime'))
        obj.serverTime = ApiClient.convertToType(data['serverTime'], 'String');
      if (data.hasOwnProperty('msgId'))
        obj.msgId = ApiClient.convertToType(data['msgId'], 'String');
      if (data.hasOwnProperty('status'))
        obj.status = ApiClient.convertToType(data['status'], 'String');
      if (data.hasOwnProperty('statusMessage'))
        obj.statusMessage = ApiClient.convertToType(data['statusMessage'], 'String');
      if (data.hasOwnProperty('sessionToken'))
        obj.sessionToken = ApiClient.convertToType(data['sessionToken'], 'String');
      if (data.hasOwnProperty('accountID'))
        obj.accountID = ApiClient.convertToType(data['accountID'], 'String');
      if (data.hasOwnProperty('accountName'))
        obj.accountName = ApiClient.convertToType(data['accountName'], 'String');
      if (data.hasOwnProperty('exchangeList'))
        obj.exchangeList = ApiClient.convertToType(data['exchangeList'], ['String']);
      if (data.hasOwnProperty('orderTypeList'))
        obj.orderTypeList = ApiClient.convertToType(data['orderTypeList'], ['String']);
      if (data.hasOwnProperty('productList'))
        obj.productList = ApiClient.convertToType(data['productList'], ['String']);
    }
    return obj;
  }

  /**
   * Time at Server.
   * @member {String} serverTime
   */
  exports.prototype.serverTime = undefined;

  /**
   * This is a unique identifier for every request into the system. Please quote this identifier to the support team if you face issues with the API request.
   * @member {String} msgId
   */
  exports.prototype.msgId = undefined;

  /**
   * Login Success or Failure.
   * @member {String} status
   */
  exports.prototype.status = undefined;

  /**
   * status Message
   * @member {String} statusMessage
   */
  exports.prototype.statusMessage = undefined;

  /**
   * Session token that should be used with API request is given in the response. It is used as an authenticator. A session token is valid for 24 hours or until a new login request is given, where you a new session token will be generated and the previous one will get expired.
   * @member {String} sessionToken
   */
  exports.prototype.sessionToken = undefined;

  /**
   * Same as Client Code for SAMCO Trading Account, is unique and used for identification of user.
   * @member {String} accountID
   */
  exports.prototype.accountID = undefined;

  /**
   * Users full name as registered with Samco Trading Account.
   * @member {String} accountName
   */
  exports.prototype.accountName = undefined;

  /**
   * List of exchanges enabled for trading for the user.
   * @member {Array.<String>} exchangeList
   */
  exports.prototype.exchangeList = undefined;

  /**
   * List of order types enabled for trading for the user.The list can be from of the following, MKT - Market Order,L- Limit Order,SL - Stop Loss Limit,SL-M - Stop loss market.
   * @member {Array.<String>} orderTypeList
   */
  exports.prototype.orderTypeList = undefined;

  /**
   * List of product types enabled for the user . It can be CNC (Cash and Carry),BO (Bracket Order),CO (Cover Order),NRML (Normal),MIS (Intraday).
   * @member {Array.<String>} productList
   */
  exports.prototype.productList = undefined;

  return exports;

}));
