// stock_note_api_documentation/src/model/ModifyOrderRequest.js
/*
 * StockNote API Documentation
 *  StockNote API is a set of Rest APIs using which users can build customized applications based on their trading requirements. It facilitates the users of the APIs to login, search symbols, place orders and execute them, view their order status, positions and holdings etc. This documentation provides you with all the necessary details to understand the SAMCO StockNote API collection. APIs are compatible with both Javascript and Python.    For any issues or support, please raise ticket to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.      For Reference you can download postman collection <a href=\"http://developers.stocknote.com/doc/StockNoteApi.postman_collection.json\">Click here</a>      For downloading a list of all tradeable scrips across exchanges please <a href=\"https://developers.stocknote.com/doc/ScripMaster.csv\">Click here</a> This is a CSV file which you can import into your database.     If you are using Python or Java as client platform to code your strategies and integrate with our APIs, you can use our StockNoteBridge  which provides pre-written wrappers over StockNote APIs that ensure seamless and reliable connectivity to the APIs and help you fast-track your development process.  <table style=\"width:100%\"><tr><th>Languages</th><th>Documentation</th></tr><tr><td>Python</td><td><a href=\"https://github.com/samco-sdk/Python-SDK/blob/master/README.md\">Python SDK Documentation</a></td></tr><tr><td>Java</td><td><a href=\"https://github.com/samco-sdk/Java-SDK/blob/master/README.md\">Java SDK Documentation</a></td></tr></table> Add 'session token' of login api response as 'x-session-token' in header param of all Apis.      NOTE: To ensure stability and there by provide seamless services to our customers, Samco may set limits on your use of the StockNote APIs (for example, limit on the number of requests sent to a specific API) . If you have additional questions regarding the rate limits on APIs, please reach out to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.   
 *
 * OpenAPI spec version: 1.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.16
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.StockNoteApiDocumentation) {
      root.StockNoteApiDocumentation = {};
    }
    root.StockNoteApiDocumentation.ModifyOrderRequest = factory(root.StockNoteApiDocumentation.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The ModifyOrderRequest model module.
   * @module model/ModifyOrderRequest
   * @version 1.0
   */

  /**
   * Constructs a new <code>ModifyOrderRequest</code>.
   * Request object for Modify an Order.
   * @alias module:model/ModifyOrderRequest
   * @class
   * @param orderValidity {String} Order validity can be DAY or IOC
   * @param parentOrderId {String} ParentOrderId required to modify an order (applicable for BO orders only)
   */
  var exports = function(orderValidity, parentOrderId) {
    this.orderValidity = orderValidity;
    this.parentOrderId = parentOrderId;
  };

  /**
   * Constructs a <code>ModifyOrderRequest</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/ModifyOrderRequest} obj Optional instance to populate.
   * @return {module:model/ModifyOrderRequest} The populated <code>ModifyOrderRequest</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();
      if (data.hasOwnProperty('orderType'))
        obj.orderType = ApiClient.convertToType(data['orderType'], 'String');
      if (data.hasOwnProperty('quantity'))
        obj.quantity = ApiClient.convertToType(data['quantity'], 'String');
      if (data.hasOwnProperty('disclosedQuantity'))
        obj.disclosedQuantity = ApiClient.convertToType(data['disclosedQuantity'], 'String');
      if (data.hasOwnProperty('orderValidity'))
        obj.orderValidity = ApiClient.convertToType(data['orderValidity'], 'String');
      if (data.hasOwnProperty('price'))
        obj.price = ApiClient.convertToType(data['price'], 'String');
      if (data.hasOwnProperty('triggerPrice'))
        obj.triggerPrice = ApiClient.convertToType(data['triggerPrice'], 'String');
      if (data.hasOwnProperty('parentOrderId'))
        obj.parentOrderId = ApiClient.convertToType(data['parentOrderId'], 'String');
      if (data.hasOwnProperty('marketProtection'))
        obj.marketProtection = ApiClient.convertToType(data['marketProtection'], 'String');
    }
    return obj;
  }

  /**
   * Type of order user has placed . It can be one of the following, MKT - Market Order,L- Limit Order,SL - Stop Loss Limit,SL-M - Stop loss market.
   * @member {String} orderType
   */
  exports.prototype.orderType = undefined;

  /**
   * Quantity of the order user wants to modify
   * @member {String} quantity
   */
  exports.prototype.quantity = undefined;

  /**
   * Quantity to disclose publicly 
   * @member {String} disclosedQuantity
   */
  exports.prototype.disclosedQuantity = undefined;

  /**
   * Order validity can be DAY or IOC
   * @member {String} orderValidity
   */
  exports.prototype.orderValidity = undefined;

  /**
   * Price at which the order was placed
   * @member {String} price
   */
  exports.prototype.price = undefined;

  /**
   * The price at which an order should be triggered in case of SL, SL-M.
   * @member {String} triggerPrice
   */
  exports.prototype.triggerPrice = undefined;

  /**
   * ParentOrderId required to modify an order (applicable for BO orders only)
   * @member {String} parentOrderId
   */
  exports.prototype.parentOrderId = undefined;

  /**
   * Percentage of MarketProtection required for ordertype MKT/SL-M to limit loss due to market price changes against the price with which order is placed. Default value is 3%.
   * @member {String} marketProtection
   */
  exports.prototype.marketProtection = undefined;

  return exports;

}));
