// stock_note_api_documentation/src/model/PositionConversionRequest.js
/*
 * StockNote API Documentation
 *  StockNote API is a set of Rest APIs using which users can build customized applications based on their trading requirements. It facilitates the users of the APIs to login, search symbols, place orders and execute them, view their order status, positions and holdings etc. This documentation provides you with all the necessary details to understand the SAMCO StockNote API collection. APIs are compatible with both Javascript and Python.    For any issues or support, please raise ticket to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.      For Reference you can download postman collection <a href=\"http://developers.stocknote.com/doc/StockNoteApi.postman_collection.json\">Click here</a>      For downloading a list of all tradeable scrips across exchanges please <a href=\"https://developers.stocknote.com/doc/ScripMaster.csv\">Click here</a> This is a CSV file which you can import into your database.     If you are using Python or Java as client platform to code your strategies and integrate with our APIs, you can use our StockNoteBridge  which provides pre-written wrappers over StockNote APIs that ensure seamless and reliable connectivity to the APIs and help you fast-track your development process.  <table style=\"width:100%\"><tr><th>Languages</th><th>Documentation</th></tr><tr><td>Python</td><td><a href=\"https://github.com/samco-sdk/Python-SDK/blob/master/README.md\">Python SDK Documentation</a></td></tr><tr><td>Java</td><td><a href=\"https://github.com/samco-sdk/Java-SDK/blob/master/README.md\">Java SDK Documentation</a></td></tr></table> Add 'session token' of login api response as 'x-session-token' in header param of all Apis.      NOTE: To ensure stability and there by provide seamless services to our customers, Samco may set limits on your use of the StockNote APIs (for example, limit on the number of requests sent to a specific API) . If you have additional questions regarding the rate limits on APIs, please reach out to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.   
 *
 * OpenAPI spec version: 1.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.16
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.StockNoteApiDocumentation) {
      root.StockNoteApiDocumentation = {};
    }
    root.StockNoteApiDocumentation.PositionConversionRequest = factory(root.StockNoteApiDocumentation.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The PositionConversionRequest model module.
   * @module model/PositionConversionRequest
   * @version 1.0
   */

  /**
   * Constructs a new <code>PositionConversionRequest</code>.
   * Request object for User Position Conversion.
   * @alias module:model/PositionConversionRequest
   * @class
   * @param symbolName {String} Symbol name of the scrip
   * @param exchange {String} Name of the exchange. If the user does not provide an exchange name, by default considered as NSE
   * @param transactionType {String} Transaction type can be either BUY or SELL
   * @param positionType {String} DAY  or NET 
   * @param netQuantity {String} Total quantity of the position
   * @param quantityToConvert {String} Quantity to be converted. Can be less than or equal to netQuantity
   * @param fromProductType {String} The existing product type of the position . It can be CNC (Cash and Carry),NRML (Normal),MIS (Intraday)
   * @param toProductType {String} The target product type user wants to convert to . It can be CNC (Cash and Carry),NRML (Normal),MIS (Intraday)
   */
  var exports = function(symbolName, exchange, transactionType, positionType, netQuantity, quantityToConvert, fromProductType, toProductType) {
    this.symbolName = symbolName;
    this.exchange = exchange;
    this.transactionType = transactionType;
    this.positionType = positionType;
    this.netQuantity = netQuantity;
    this.quantityToConvert = quantityToConvert;
    this.fromProductType = fromProductType;
    this.toProductType = toProductType;
  };

  /**
   * Constructs a <code>PositionConversionRequest</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/PositionConversionRequest} obj Optional instance to populate.
   * @return {module:model/PositionConversionRequest} The populated <code>PositionConversionRequest</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();
      if (data.hasOwnProperty('symbolName'))
        obj.symbolName = ApiClient.convertToType(data['symbolName'], 'String');
      if (data.hasOwnProperty('exchange'))
        obj.exchange = ApiClient.convertToType(data['exchange'], 'String');
      if (data.hasOwnProperty('transactionType'))
        obj.transactionType = ApiClient.convertToType(data['transactionType'], 'String');
      if (data.hasOwnProperty('positionType'))
        obj.positionType = ApiClient.convertToType(data['positionType'], 'String');
      if (data.hasOwnProperty('netQuantity'))
        obj.netQuantity = ApiClient.convertToType(data['netQuantity'], 'String');
      if (data.hasOwnProperty('quantityToConvert'))
        obj.quantityToConvert = ApiClient.convertToType(data['quantityToConvert'], 'String');
      if (data.hasOwnProperty('fromProductType'))
        obj.fromProductType = ApiClient.convertToType(data['fromProductType'], 'String');
      if (data.hasOwnProperty('toProductType'))
        obj.toProductType = ApiClient.convertToType(data['toProductType'], 'String');
    }
    return obj;
  }

  /**
   * Symbol name of the scrip
   * @member {String} symbolName
   */
  exports.prototype.symbolName = undefined;

  /**
   * Name of the exchange. If the user does not provide an exchange name, by default considered as NSE
   * @member {String} exchange
   */
  exports.prototype.exchange = undefined;

  /**
   * Transaction type can be either BUY or SELL
   * @member {String} transactionType
   */
  exports.prototype.transactionType = undefined;

  /**
   * DAY  or NET 
   * @member {String} positionType
   */
  exports.prototype.positionType = undefined;

  /**
   * Total quantity of the position
   * @member {String} netQuantity
   */
  exports.prototype.netQuantity = undefined;

  /**
   * Quantity to be converted. Can be less than or equal to netQuantity
   * @member {String} quantityToConvert
   */
  exports.prototype.quantityToConvert = undefined;

  /**
   * The existing product type of the position . It can be CNC (Cash and Carry),NRML (Normal),MIS (Intraday)
   * @member {String} fromProductType
   */
  exports.prototype.fromProductType = undefined;

  /**
   * The target product type user wants to convert to . It can be CNC (Cash and Carry),NRML (Normal),MIS (Intraday)
   * @member {String} toProductType
   */
  exports.prototype.toProductType = undefined;

  return exports;

}));
