// stock_note_api_documentation/src/model/ScripDetails.js
/*
 * StockNote API Documentation
 *  StockNote API is a set of Rest APIs using which users can build customized applications based on their trading requirements. It facilitates the users of the APIs to login, search symbols, place orders and execute them, view their order status, positions and holdings etc. This documentation provides you with all the necessary details to understand the SAMCO StockNote API collection. APIs are compatible with both Javascript and Python.    For any issues or support, please raise ticket to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.      For Reference you can download postman collection <a href=\"http://developers.stocknote.com/doc/StockNoteApi.postman_collection.json\">Click here</a>      For downloading a list of all tradeable scrips across exchanges please <a href=\"https://developers.stocknote.com/doc/ScripMaster.csv\">Click here</a> This is a CSV file which you can import into your database.     If you are using Python or Java as client platform to code your strategies and integrate with our APIs, you can use our StockNoteBridge  which provides pre-written wrappers over StockNote APIs that ensure seamless and reliable connectivity to the APIs and help you fast-track your development process.  <table style=\"width:100%\"><tr><th>Languages</th><th>Documentation</th></tr><tr><td>Python</td><td><a href=\"https://github.com/samco-sdk/Python-SDK/blob/master/README.md\">Python SDK Documentation</a></td></tr><tr><td>Java</td><td><a href=\"https://github.com/samco-sdk/Java-SDK/blob/master/README.md\">Java SDK Documentation</a></td></tr></table> Add 'session token' of login api response as 'x-session-token' in header param of all Apis.      NOTE: To ensure stability and there by provide seamless services to our customers, Samco may set limits on your use of the StockNote APIs (for example, limit on the number of requests sent to a specific API) . If you have additional questions regarding the rate limits on APIs, please reach out to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.   
 *
 * OpenAPI spec version: 1.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.16
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.StockNoteApiDocumentation) {
      root.StockNoteApiDocumentation = {};
    }
    root.StockNoteApiDocumentation.ScripDetails = factory(root.StockNoteApiDocumentation.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The ScripDetails model module.
   * @module model/ScripDetails
   * @version 1.0
   */

  /**
   * Constructs a new <code>ScripDetails</code>.
   * @alias module:model/ScripDetails
   * @class
   * @param exchange {String} Name of the exchange.Valid exchanges values (BSE/ NSE/ NFO/ MCX/ CDS).If the user does not provide an exchange name, by default considered as NSE.For trading with BSE, NFO, CDS and MCX, exchange is mandatory.
   * @param scripDescription {String} scrip description 
   * @param tradingSymbol {String} Trading Symbol of the scrip.
   * @param isin {String} The standard ISIN representing stocks uniquely at international level. It is same for every exchange.
   * @param tickSize {String} The value of a single price tick. Default value is 0.05
   * @param instrument {String} Instrument Name.
   * @param quantityInLots {String} Lot size of the symbol to be traded. At the time of placing order, the quantity should be in multiples of Broadlot Qty only.
   */
  var exports = function(exchange, scripDescription, tradingSymbol, isin, tickSize, instrument, quantityInLots) {
    this.exchange = exchange;
    this.scripDescription = scripDescription;
    this.tradingSymbol = tradingSymbol;
    this.isin = isin;
    this.tickSize = tickSize;
    this.instrument = instrument;
    this.quantityInLots = quantityInLots;
  };

  /**
   * Constructs a <code>ScripDetails</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/ScripDetails} obj Optional instance to populate.
   * @return {module:model/ScripDetails} The populated <code>ScripDetails</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();
      if (data.hasOwnProperty('exchange'))
        obj.exchange = ApiClient.convertToType(data['exchange'], 'String');
      if (data.hasOwnProperty('scripDescription'))
        obj.scripDescription = ApiClient.convertToType(data['scripDescription'], 'String');
      if (data.hasOwnProperty('tradingSymbol'))
        obj.tradingSymbol = ApiClient.convertToType(data['tradingSymbol'], 'String');
      if (data.hasOwnProperty('isin'))
        obj.isin = ApiClient.convertToType(data['isin'], 'String');
      if (data.hasOwnProperty('bodLotQuantity'))
        obj.bodLotQuantity = ApiClient.convertToType(data['bodLotQuantity'], 'String');
      if (data.hasOwnProperty('tickSize'))
        obj.tickSize = ApiClient.convertToType(data['tickSize'], 'String');
      if (data.hasOwnProperty('instrument'))
        obj.instrument = ApiClient.convertToType(data['instrument'], 'String');
      if (data.hasOwnProperty('quantityInLots'))
        obj.quantityInLots = ApiClient.convertToType(data['quantityInLots'], 'String');
    }
    return obj;
  }

  /**
   * Name of the exchange.Valid exchanges values (BSE/ NSE/ NFO/ MCX/ CDS).If the user does not provide an exchange name, by default considered as NSE.For trading with BSE, NFO, CDS and MCX, exchange is mandatory.
   * @member {String} exchange
   */
  exports.prototype.exchange = undefined;

  /**
   * scrip description 
   * @member {String} scripDescription
   */
  exports.prototype.scripDescription = undefined;

  /**
   * Trading Symbol of the scrip.
   * @member {String} tradingSymbol
   */
  exports.prototype.tradingSymbol = undefined;

  /**
   * The standard ISIN representing stocks uniquely at international level. It is same for every exchange.
   * @member {String} isin
   */
  exports.prototype.isin = undefined;

  /**
   * @member {String} bodLotQuantity
   */
  exports.prototype.bodLotQuantity = undefined;

  /**
   * The value of a single price tick. Default value is 0.05
   * @member {String} tickSize
   */
  exports.prototype.tickSize = undefined;

  /**
   * Instrument Name.
   * @member {String} instrument
   */
  exports.prototype.instrument = undefined;

  /**
   * Lot size of the symbol to be traded. At the time of placing order, the quantity should be in multiples of Broadlot Qty only.
   * @member {String} quantityInLots
   */
  exports.prototype.quantityInLots = undefined;

  return exports;

}));
