// stock_note_api_documentation/src/model/StreamingMarkDataResponse.js
/*
 * StockNote API Documentation
 *  StockNote API is a set of Rest APIs using which users can build customized applications based on their trading requirements. It facilitates the users of the APIs to login, search symbols, place orders and execute them, view their order status, positions and holdings etc. This documentation provides you with all the necessary details to understand the SAMCO StockNote API collection. APIs are compatible with both Javascript and Python.    For any issues or support, please raise ticket to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.      For Reference you can download postman collection <a href=\"http://developers.stocknote.com/doc/StockNoteApi.postman_collection.json\">Click here</a>      For downloading a list of all tradeable scrips across exchanges please <a href=\"https://developers.stocknote.com/doc/ScripMaster.csv\">Click here</a> This is a CSV file which you can import into your database.     If you are using Python or Java as client platform to code your strategies and integrate with our APIs, you can use our StockNoteBridge  which provides pre-written wrappers over StockNote APIs that ensure seamless and reliable connectivity to the APIs and help you fast-track your development process.  <table style=\"width:100%\"><tr><th>Languages</th><th>Documentation</th></tr><tr><td>Python</td><td><a href=\"https://github.com/samco-sdk/Python-SDK/blob/master/README.md\">Python SDK Documentation</a></td></tr><tr><td>Java</td><td><a href=\"https://github.com/samco-sdk/Java-SDK/blob/master/README.md\">Java SDK Documentation</a></td></tr></table> Add 'session token' of login api response as 'x-session-token' in header param of all Apis.      NOTE: To ensure stability and there by provide seamless services to our customers, Samco may set limits on your use of the StockNote APIs (for example, limit on the number of requests sent to a specific API) . If you have additional questions regarding the rate limits on APIs, please reach out to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.   
 *
 * OpenAPI spec version: 1.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.16
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/AskValues', 'model/BidValues'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./AskValues'), require('./BidValues'));
  } else {
    // Browser globals (root is window)
    if (!root.StockNoteApiDocumentation) {
      root.StockNoteApiDocumentation = {};
    }
    root.StockNoteApiDocumentation.StreamingMarkDataResponse = factory(root.StockNoteApiDocumentation.ApiClient, root.StockNoteApiDocumentation.AskValues, root.StockNoteApiDocumentation.BidValues);
  }
}(this, function(ApiClient, AskValues, BidValues) {
  'use strict';

  /**
   * The StreamingMarkDataResponse model module.
   * @module model/StreamingMarkDataResponse
   * @version 1.0
   */

  /**
   * Constructs a new <code>StreamingMarkDataResponse</code>.
   * @alias module:model/StreamingMarkDataResponse
   * @class
   * @param askValues {Array.<module:model/AskValues>} Get List of AskValues .
   * @param bidValues {Array.<module:model/BidValues>} Get List of BidValues .
   * @param ltt {String} Last transaction time in milliseconds
   * @param lttUTC {String} Last transaction time in UTC time zone format
   * @param sym {String} Actual symbol name of the scrip
   * @param tBQ {String} Total quantity of BUY transaction
   * @param tSQ {String} Total quantity of SELL transaction
   */
  var exports = function(askValues, bidValues, ltt, lttUTC, sym, tBQ, tSQ) {
    this.askValues = askValues;
    this.bidValues = bidValues;
    this.ltt = ltt;
    this.lttUTC = lttUTC;
    this.sym = sym;
    this.tBQ = tBQ;
    this.tSQ = tSQ;
  };

  /**
   * Constructs a <code>StreamingMarkDataResponse</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/StreamingMarkDataResponse} obj Optional instance to populate.
   * @return {module:model/StreamingMarkDataResponse} The populated <code>StreamingMarkDataResponse</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();
      if (data.hasOwnProperty('askValues'))
        obj.askValues = ApiClient.convertToType(data['askValues'], [AskValues]);
      if (data.hasOwnProperty('bidValues'))
        obj.bidValues = ApiClient.convertToType(data['bidValues'], [BidValues]);
      if (data.hasOwnProperty('ltt'))
        obj.ltt = ApiClient.convertToType(data['ltt'], 'String');
      if (data.hasOwnProperty('lttUTC'))
        obj.lttUTC = ApiClient.convertToType(data['lttUTC'], 'String');
      if (data.hasOwnProperty('sym'))
        obj.sym = ApiClient.convertToType(data['sym'], 'String');
      if (data.hasOwnProperty('tBQ'))
        obj.tBQ = ApiClient.convertToType(data['tBQ'], 'String');
      if (data.hasOwnProperty('tSQ'))
        obj.tSQ = ApiClient.convertToType(data['tSQ'], 'String');
    }
    return obj;
  }

  /**
   * Get List of AskValues .
   * @member {Array.<module:model/AskValues>} askValues
   */
  exports.prototype.askValues = undefined;

  /**
   * Get List of BidValues .
   * @member {Array.<module:model/BidValues>} bidValues
   */
  exports.prototype.bidValues = undefined;

  /**
   * Last transaction time in milliseconds
   * @member {String} ltt
   */
  exports.prototype.ltt = undefined;

  /**
   * Last transaction time in UTC time zone format
   * @member {String} lttUTC
   */
  exports.prototype.lttUTC = undefined;

  /**
   * Actual symbol name of the scrip
   * @member {String} sym
   */
  exports.prototype.sym = undefined;

  /**
   * Total quantity of BUY transaction
   * @member {String} tBQ
   */
  exports.prototype.tBQ = undefined;

  /**
   * Total quantity of SELL transaction
   * @member {String} tSQ
   */
  exports.prototype.tSQ = undefined;

  return exports;

}));
