// stock_note_api_documentation/src/model/StreamingQuoteDataRequest.js
/*
 * StockNote API Documentation
 *  StockNote API is a set of Rest APIs using which users can build customized applications based on their trading requirements. It facilitates the users of the APIs to login, search symbols, place orders and execute them, view their order status, positions and holdings etc. This documentation provides you with all the necessary details to understand the SAMCO StockNote API collection. APIs are compatible with both Javascript and Python.    For any issues or support, please raise ticket to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.      For Reference you can download postman collection <a href=\"http://developers.stocknote.com/doc/StockNoteApi.postman_collection.json\">Click here</a>      For downloading a list of all tradeable scrips across exchanges please <a href=\"https://developers.stocknote.com/doc/ScripMaster.csv\">Click here</a> This is a CSV file which you can import into your database.     If you are using Python or Java as client platform to code your strategies and integrate with our APIs, you can use our StockNoteBridge  which provides pre-written wrappers over StockNote APIs that ensure seamless and reliable connectivity to the APIs and help you fast-track your development process.  <table style=\"width:100%\"><tr><th>Languages</th><th>Documentation</th></tr><tr><td>Python</td><td><a href=\"https://github.com/samco-sdk/Python-SDK/blob/master/README.md\">Python SDK Documentation</a></td></tr><tr><td>Java</td><td><a href=\"https://github.com/samco-sdk/Java-SDK/blob/master/README.md\">Java SDK Documentation</a></td></tr></table> Add 'session token' of login api response as 'x-session-token' in header param of all Apis.      NOTE: To ensure stability and there by provide seamless services to our customers, Samco may set limits on your use of the StockNote APIs (for example, limit on the number of requests sent to a specific API) . If you have additional questions regarding the rate limits on APIs, please reach out to us using the <a href=\"https://www.samco.in/support/index/tradeapi\">support link</a> and we will be happy to assist you.   
 *
 * OpenAPI spec version: 1.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.16
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.StockNoteApiDocumentation) {
      root.StockNoteApiDocumentation = {};
    }
    root.StockNoteApiDocumentation.StreamingQuoteDataRequest = factory(root.StockNoteApiDocumentation.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The StreamingQuoteDataRequest model module.
   * @module model/StreamingQuoteDataRequest
   * @version 1.0
   */

  /**
   * Constructs a new <code>StreamingQuoteDataRequest</code>.
   * Request object for Streaming live market data.
   * @alias module:model/StreamingQuoteDataRequest
   * @class
   * @param streamingType {String} Streaming type.Added for future use.Pass this value as “quote” always
   * @param symbols {Array.<String>} List of symbols contains listing nos used to get quote.
   * @param requestType {String} Request type.This can be Subscribe/Unsubscribe. Subscribe - Receive continuous quote data. UnSubscribe - Stop receiving the continuous quote data. *It's important to unsubscribe when the streaming is no longer needed.  
   * @param responseFormat {String} Response data format. Currently supports json only
   */
  var exports = function(streamingType, symbols, requestType, responseFormat) {
    this.streamingType = streamingType;
    this.symbols = symbols;
    this.requestType = requestType;
    this.responseFormat = responseFormat;
  };

  /**
   * Constructs a <code>StreamingQuoteDataRequest</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/StreamingQuoteDataRequest} obj Optional instance to populate.
   * @return {module:model/StreamingQuoteDataRequest} The populated <code>StreamingQuoteDataRequest</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();
      if (data.hasOwnProperty('streaming_type'))
        obj.streamingType = ApiClient.convertToType(data['streaming_type'], 'String');
      if (data.hasOwnProperty('symbols'))
        obj.symbols = ApiClient.convertToType(data['symbols'], ['String']);
      if (data.hasOwnProperty('request_type'))
        obj.requestType = ApiClient.convertToType(data['request_type'], 'String');
      if (data.hasOwnProperty('response_format'))
        obj.responseFormat = ApiClient.convertToType(data['response_format'], 'String');
    }
    return obj;
  }

  /**
   * Streaming type.Added for future use.Pass this value as “quote” always
   * @member {String} streamingType
   */
  exports.prototype.streamingType = undefined;

  /**
   * List of symbols contains listing nos used to get quote.
   * @member {Array.<String>} symbols
   */
  exports.prototype.symbols = undefined;

  /**
   * Request type.This can be Subscribe/Unsubscribe. Subscribe - Receive continuous quote data. UnSubscribe - Stop receiving the continuous quote data. *It's important to unsubscribe when the streaming is no longer needed.  
   * @member {String} requestType
   */
  exports.prototype.requestType = undefined;

  /**
   * Response data format. Currently supports json only
   * @member {String} responseFormat
   */
  exports.prototype.responseFormat = undefined;

  return exports;

}));
